flow360/component/simulation/translator/solver_translator.py

```diff
@@ -0,0 +1,180 @@
+import re
+from typing import Union
+
+from flow360.component.simulation.models.surface_models import (
+    Freestream,
+    SlipWall,
+    SymmetryPlane,
+    Wall,
+)
+from flow360.component.simulation.models.volume_models import Fluid
+from flow360.component.simulation.outputs.outputs import (
+    SliceOutput,
+    SurfaceOutput,
+    TimeAverageVolumeOutput,
+    VolumeOutput,
+)
+from flow360.component.simulation.simulation_params import SimulationParams
+from flow360.component.simulation.translator.utils import (
+    get_attribute_from_first_instance,
+    preprocess_input,
+)
+from flow360.component.simulation.unit_system import LengthType
+
+
+def to_dict(input_params):
+    return input_params.model_dump(by_alias=True, exclude_none=True)
+
+
+def to_dict_without_unit(input_params):
+    unit_keys = {"value", "units"}
+    output_dict = to_dict(input_params)
+    for key, value in output_dict.items():
+        output_dict[key] = (
+            value["value"] if isinstance(value, dict) and value.keys() == unit_keys else value
+        )
+    return output_dict
+
+
+def remove_empty_keys(input_dict):
+    # TODO: implement
+    return input_dict
+
+
+def init_output_attr_dict(obj_list, class_type):
+    return {
+        "animationFrequency": get_attribute_from_first_instance(obj_list, class_type, "frequency"),
+        "animationFrequencyOffset": get_attribute_from_first_instance(
+            obj_list, class_type, "frequency_offset"
+        ),
+        "outputFormat": get_attribute_from_first_instance(obj_list, class_type, "output_format"),
+    }
+
+
+def get_output_fields(obj_list, class_type):
```

Comment:
I added this before I realized surface and slice won't use it for now, but for prepopulation maybe in the future? There's no harm in keeping a function right

Reply:
What do you mean by prepopulation in the future?

From my understanding this only works for volumeOutput and we do not have many "volumeOutput" so there is not much point moving stuff in to a function that is only going to be used once. 

Maybe I am lazy but I prefer having inline codes rather than jumping around to see what a function actually does (especially if that function is only used once).

Also the name should then be get_output_fields_from_first_instance.

Reply:
Maybe we want to use this function to prepopulation output_fields in the future. It's ok I'll remove it